capa/features/common.py

```diff
@@ -272,6 +272,42 @@ def __str__(self):
         return f'substring("{self.value}", matches = {matches})'
 
 
+class Registry(String):
+    # todo: add a way to tell whether this registry key was created, accessed, or deleted.
+    def __init__(self, value: str, description=None):
+        super().__init__(value, description)
+
+    def __eq__(self, other):
+        # Registry instance is in a ruleset
```

Comment:
```suggestion
```

Reply:
`__eq__` compares two objects

Reply:
I think the comment was unclear. what I am trying to say is that given that capa matches features in the following manner: RuleSetFeature ==? ExtractedFeature, then it's the eq method of the RuleSetFeature that's going to be called always, so based on that we can assume (i believe?) that if the eq method of a Registry instance is the one being called then that rule feature is a registry, and we should be selective and only match against other registry keys (since that's what the rule author wants).
Is my reasoning correct?

Reply:
the capa feature matcher primarily uses hash lookups to decide if features are present, so features need to provide a meaningful hash and eq implementation. usually you can rely on the implementations here: https://github.com/mandiant/capa/blob/7ef78fdbce3959529eed5d8bd96b67348838f6e7/capa/features/common.py#L120-L124

these rely on the feature name and values being distinct, which i think they would be for this Registry feature.

Reply:
i *think* we can probably get away with:

```py
class Registry(String):
    pass
```

Reply:
ah, I missed that the feature checks for both the value as well as the class name. do you think the desired feature (string matching against both string and regkey, and regkey against only regkey features) is worth implementing?

other ideas that come to mind for implementing this include:
- modifying the __eq__ method of the String feature.
- setting the name of a Registry instance to "string", which seems wrong...

I'll try to think of other alternatives if this is worth having.

Reply:
i'd recommend moving this discussion to a dedicated github issue where we can figure out the feature, the rule syntax, how to handle the operations (read/write/delete/etc.), etc.